### PR TITLE
login-page-connection

### PR DESCRIPTION
**Pull request by ABabayan32**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
   },
   "dependencies": {
     "vue": "^3.2.45",
+    "vue-cookie": "^1.1.4",
```
Comment (lavrovpy) [inline package.json]: It is not necessary to use vue-cookie here. We can achieve the same using only built-in tools https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie

This is just FYI, no need to change anything

File: src/static_functions.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+function abortFetching() {
```
Comment (lavrovpy) [inline src/static_functions.js]: it seems that you do not use abortFetching anywhere. Usually, there is no need to abort fetch in most cases unless the backend takes too much time to respond. Here are more details https://stackoverflow.com/questions/31061838/how-do-i-cancel-an-http-fetch-request

File: src/static_functions.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    console.log('Now aborting');
+    controller.abort()
+}
+
+function getHeaders() {
+    return {
+         "Access-Control-Allow-Origin": "*", 'Access-Control-Allow-Methods':'GET,PUT,POST,DELETE,PATCH,OPTIONS'}
```
Comment (lavrovpy) [inline src/static_functions.js]: nice, but better to write every key:value pair from a new line, like

```
return {
    "Access-Control-Allow-Origin": "*",
    'Access-Control-Allow-Methods':'GET,PUT,POST,DELETE,PATCH,OPTIONS'
}
```

and use quotes consistently

File: src/static_functions.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+function getAuthenticatedHeaders () {
```
Comment (lavrovpy) [inline src/static_functions.js]: better to use it in this way:

```
return {
    ...getHeaders(),
    Authorization: getCookie('access_token')
}
```

File: src/static_functions.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const defaultHeaders = getHeaders();
+    defaultHeaders["Authorization"] = getCookie("access_token");
+   return defaultHeaders;
+}
+
+function getCookie(cname) {
```
Comment (lavrovpy) [inline src/static_functions.js]: I guess you can remove this function since you use vue-cookie library

File: src/static_functions.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    var name = cname + "=";
```
Comment (lavrovpy) [inline src/static_functions.js]: do not use _var_ keyword to declare a variable

File: src/views/Login.vue [hunk 7]
```diff
@@ -1,14 +1,33 @@
...
     return {
       login: '',
       password: ''
     }
-  }
-}
+  },
+  methods: {
+    onLogin() {
+      fetch('http://localhost:8080/login', {
```
Comment (lavrovpy) [inline src/views/Login.vue]: better to move the url to a config file

File: src/views/Login.vue [hunk 8]
```diff
@@ -1,14 +1,33 @@
...
+  data() {
     return {
       login: '',
       password: ''
     }
-  }
-}
+  },
+  methods: {
+    onLogin() {
```
Comment (lavrovpy) [inline src/views/Login.vue]: better to rename onLogin --> login

File: src/views/Login.vue [hunk 9]
```diff
@@ -1,14 +1,33 @@
...
+      fetch('http://localhost:8080/login', {
+            method: 'POST',
+            node: 'cors',
+            cache: 'no-cache',
+            credentials: 'same-origin',
+            headers: {
```
Comment (lavrovpy) [inline src/views/Login.vue]: use your getHeaders() function here

File: src/views/Login.vue [hunk 10]
```diff
@@ -17,6 +36,6 @@ export default {
     <p>Log in to proceed</p>
     <input v-model="login" placeholder="login" />
     <input type="password" v-model="password" placeholder="password" />
-    <button @click="">Login</button>
+    <button @click="onLogin">Login</button>
```
Comment (lavrovpy) [inline src/views/Login.vue]: rename onLogin --> login

**Review by lavrovpy — CHANGES_REQUESTED** [review]

**Comment by ABabayan32** [comment]
Finalizing Login page

**Pull request merged**